src/app/components/receive/receive.component.ts

```diff
@@ -79,38 +56,13 @@ export class ReceiveComponent implements OnInit {
         }
       }
     }
-
   }
 
-  async loadPendingForAccount(account) {
-    this.pendingBlocks = [];
-
-    let pending;
-    if (this.settings.settings.minimumReceive) {
-      const minAmount = this.util.nano.mnanoToRaw(this.settings.settings.minimumReceive);
-      pending = await this.api.pendingLimit(account, 50, minAmount.toString(10));
-    } else {
-      pending = await this.api.pending(account, 50);
-    }
-    if (!pending || !pending.blocks) return;
-
-    for (let block in pending.blocks) {
-      const pendingTx = {
-        block: block,
-        amount: pending.blocks[block].amount,
-        source: pending.blocks[block].source,
-        account: account,
-      };
-      this.pendingBlocks.push(pendingTx);
-    }
-  }
-
-  async getPending(account) {
-    if (!account || account == 0) {
-      await this.loadPendingForAll();
-    } else {
-      await this.loadPendingForAccount(account);
-    }
+  async getPending() {
+    // clear the list of pending blocks. Updated again with reloadBalances()
+    this.walletService.clearPendingBlocks()
+    await this.walletService.reloadBalances(true)
+    await this.loadPendingForAll();
   }
```

Comment:
Don't we need the functionality to get pending for a specific account?
The old function has both functionalities.

Reply:
No because it works completely different now. Instead of calling the API to get pending for all accounts or a specific account the data is taken from the already retrieved pending in the wallet service. There is no "burden" to get the full list each time. I filter for specific account directly in the html instead.

Reply:
Alright 👌